rollup/da_syncer/batch_queue.go

```diff
@@ -85,25 +93,85 @@ func (bq *BatchQueue) addBatch(batch da.Entry) {
 	bq.batchesMap.Set(batch.BatchIndex(), heapElement)
 }
 
-// deleteBatch deletes data committed in the batch from map, because this batch is reverted or finalized
-// updates DASyncedL1BlockNumber
-func (bq *BatchQueue) deleteBatch(batch da.Entry) {
-	batchHeapElement, exists := bq.batchesMap.Get(batch.BatchIndex())
+func (bq *BatchQueue) handleRevertEvent(event l1.RollupEvent) error {
+	switch event.Type() {
+	case l1.RevertEventV0Type:
+		revertBatch, ok := event.(*l1.RevertBatchEventV0)
+		if !ok {
+			return fmt.Errorf("unexpected type of revert event: %T, expected RevertEventV0Type", event)
+		}
+
+		bq.deleteBatch(revertBatch.BatchIndex().Uint64())
+	case l1.RevertEventV7Type:
+		revertBatch, ok := event.(*l1.RevertBatchEventV7)
+		if !ok {
+			return fmt.Errorf("unexpected type of revert event: %T, expected RevertEventV7Type", event)
+		}
+
+		// delete all batches from revertBatch.StartBatchIndex (inclusive) to revertBatch.FinishBatchIndex (inclusive)
+		for i := revertBatch.StartBatchIndex().Uint64(); i <= revertBatch.FinishBatchIndex().Uint64(); i++ {
+			bq.deleteBatch(i)
+		}
+	default:
+		return fmt.Errorf("unexpected type of revert event: %T", event)
+	}
+
+	return nil
+}
+
+func (bq *BatchQueue) deleteBatch(batchIndex uint64) (deleted bool) {
+	batchHeapElement, exists := bq.batchesMap.Get(batchIndex)
 	if !exists {
-		return
+		return false
 	}
 
-	bq.batchesMap.Delete(batch.BatchIndex())
+	bq.batchesMap.Delete(batchIndex)
 	bq.batches.Remove(batchHeapElement)
 
-	// we store here min height of currently loaded batches to be able to start syncing from the same place in case of restart
-	// TODO: we should store this information when the batch is done being processed to avoid inconsistencies
-	rawdb.WriteDASyncedL1BlockNumber(bq.db, batch.L1BlockNumber()-1)
+	return true
+}
+
+// processAndDeleteBatch deletes data committed in the batch from map, because this batch is reverted or finalized
+// updates DASyncedL1BlockNumber
+func (bq *BatchQueue) processAndDeleteBatch(batch da.Entry) da.EntryWithBlocks {
+	if !bq.deleteBatch(batch.BatchIndex()) {
+		return nil
+	}
+
+	entryWithBlocks, ok := batch.(da.EntryWithBlocks)
+	if !ok {
+		// this should only happen if we delete a reverted batch
+		return nil
+	}
+
+	// sanity check that the next batch is the one we expect
+	if bq.previousBatch.BatchIndex > 0 && bq.previousBatch.BatchIndex+1 != entryWithBlocks.BatchIndex() {
+		log.Info("BatchQueue: skipping batch ", "currentBatch", entryWithBlocks.BatchIndex(), "previousBatch", bq.previousBatch.BatchIndex)
```

Comment:
Would this happen? If not, why is this `Info`?

Reply:
this might happen if we specify a wrong batch in recovery mode for example